apps/web/src/components/AssessmentStepTracker/utils.ts

```diff
@@ -188,11 +205,200 @@ export const sortResultsAndAddOrdinal = (
   return resultsWithOrdinal.sort((resultA, resultB) => {
     return (
       getBookmarkValue(resultB) - getBookmarkValue(resultA) ||
-      getDecisionValue(resultA) - getDecisionValue(resultB) ||
+      getDecisionValue(resultA.decision) - getDecisionValue(resultB.decision) ||
       getPriorityValue(resultB) - getPriorityValue(resultA) ||
       getVeteranValue(resultB) - getVeteranValue(resultA) ||
       compareLastNames(resultA, resultB) ||
       compareFirstNames(resultA, resultB)
     );
   });
 };
+
+const getResultsDecision = (
+  step: AssessmentStep,
+  results?: AssessmentResult[],
+): NullableDecision => {
+  if (!results) return NO_DECISION;
+  let hasFailure: boolean = false;
+  let hasOnHold: boolean = false;
+  let hasToAssess: boolean = false;
+
+  const stepResults = results
+    ?.filter((result) => {
+      return result.assessmentStep?.id === step.id;
+    })
+    .filter(notEmpty);
```

Comment:
```suggestion
  const stepResults = results.filter((result) => {
    return result.assessmentStep?.id === step.id;
  });
```
Typescript is smart enough to realize that, if code has made it past line 221, then results must be `AssessmentResult[]`